libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2207,13 +2207,43 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 	}
 }
 
+
+void TypeChecker::typeCheckStringConcatFunction(
+	FunctionCall const& _functionCall,
+	FunctionType const* _functionType
+)
+{
+	solAssert(_functionType, "");
+	solAssert(_functionType->kind() == FunctionType::Kind::StringConcat, "");
+	solAssert(_functionCall.names().empty(), "");
+
+	typeCheckFunctionGeneralChecks(_functionCall, _functionType);
+
+	for (shared_ptr<Expression const> const& argument: _functionCall.arguments())
+	{
+		Type const* argumentType = type(*argument);
+		bool notConvertibleToString = !argumentType->isImplicitlyConvertibleTo(*TypeProvider::stringMemory());
+		bool numberLiteral = (dynamic_cast<RationalNumberType const*>(argumentType) != nullptr);
+
+		if (notConvertibleToString || numberLiteral)
+			m_errorReporter.typeError(
+				0000_error, // I'm not how to get new error codes for this, but we'll get to it in time
```

Comment:
``scripts/error_codes.py --fix```

Reply:
You can also just come up with a random number yourself.